controllers/authController.js

```diff
@@ -0,0 +1,93 @@
+const bcrypt = require('bcrypt');
+const jwt = require('jsonwebtoken');
+const UserHistory = require('../models/UserHistory');
+const User = require('../models/Users');
+const { signupSchema, loginSchema } = require('../validations/userValidation');
+require('dotenv').config();
+
+exports.signup = async (req, res) => {
+  const { name, password, email, role } = req.body;
+  const { error } = signupSchema.validate(req.body);
+  if (error) return res.status(400).json({ message: error.details[0].message });
+  try {
+    const existingUser = await User.findOne({ where: { email } });
+    if (existingUser)
+      return res
+        .status(400)
+        .json({ message: 'User with this email already exists.' });
+
+    const newUser = await User.create({
+      name,
+      email,
+      password,
+      role,
+    });
+
+    res.status(201).json({ message: `${name} registered successfully!` });
+  } catch (err) {
+    console.error(err);
+    res.status(500).json({ message: 'Server Error' });
+  }
```

Comment:
as per our consultation previously I asked you to create a wrapper function to handle try catch.

Can you please highlight the use of it in this PR?